src/util_i_targeting.nss

```diff
@@ -0,0 +1,894 @@
+    }
+    else
+        Warning("Targeting data for target hook " + IntToString(nHookID) + " not found");
+
+    return th;
+}
+
+struct TargetingHook GetTargetingHookDataByVarName(object oPC, string sVarName)
+{
+    int nHookID = GetTargetingHookID(oPC, sVarName);
+    return GetTargetingHookDataByHookID(nHookID);
+}
+
+sqlquery GetTargetList(object oPC, string sVarName, int nIndex = -1)
+{
+    string s =  "SELECT sTargetObject, sTargetArea, vTargetLocation " +
+                "FROM targeting_targets " +
+                "WHERE sUUID = @sUUID " +
+                    "AND sVarName = @sVarName" +
+                    (nIndex == -1 ? ";" : "LIMIT 1 OFFSET " + IntToString(nIndex)) + ";";
+
+    sqlquery q = _PrepareTargetingQuery(s);
+    SqlBindString(q, "@sUUID", GetObjectUUID(oPC));
+    SqlBindString(q, "@sVarName", sVarName);
+
+    return q;
+}
+
+int AddTargetToTargetList(object oPC, string sVarName, object oTarget, object oArea, vector vTarget)
+{
+    string s =  "INSERT INTO targeting_targets (sUUID, sVarName, sTargetObject, sTargetArea, vTargetLocation) " +
+                "VALUES (@sUUID, @sVarName, @sTargetObject, @sTargetArea, @vTargetLocation);";
+
+    sqlquery q = _PrepareTargetingQuery(s);
+    SqlBindString(q, "@sUUID", GetObjectUUID(oPC));
+    SqlBindString(q, "@sVarName", sVarName);
+    SqlBindString(q, "@sTargetObject", ObjectToString(oTarget));
+    SqlBindString(q, "@sTargetArea", ObjectToString(oArea));
+    SqlBindVector(q, "@vTargetLocation", vTarget);
+    SqlStep(q);
+
+    return CountTargetingHookTargets(oPC, sVarName);
+}
+
+void DeleteTargetList(object oPC, string sVarName)
+{
+    string s =  "DELETE FROM targeting_targets " +
+                "WHERE sUUID = @sUUID " +
+                    "AND sVarName = @sVarName;";
+
+    sqlquery q = _PrepareTargetingQuery(s);
+    SqlBindString(q, "@sUUID", GetObjectUUID(oPC));
+    SqlBindString(q, "@sVarName", sVarName);
+
+    SqlStep(q);
+}
+
+void EnterTargetingModeByHookID(int nHookID, int nBehavior = TARGET_BEHAVIOR_ADD)
+{
+    struct TargetingHook th = GetTargetingHookDataByHookID(nHookID);
+
+    if (th == TARGETING_HOOK_INVALID)
+    {
+        Warning("EnterTargetingModeByHookID::Unable to retrieve valid targeting data for " +
+            "targeting hook " + IntToString(nHookID));
+        return;
+    }
+
+    if (GetIsObjectValid(th.oPC))
+        _EnterTargetingMode(th.oPC, th.nObjectType, nHookID, nBehavior);
+}
+
+void EnterTargetingModeByVarName(object oPC, string sVarName, int nBehavior = TARGET_BEHAVIOR_ADD)
+{
+    struct TargetingHook th = GetTargetingHookDataByVarName(oPC, sVarName);
+
+    if (th == TARGETING_HOOK_INVALID)
+    {
+        Warning("EnterTargetingModeByVarName::Unable to retrieve valid targeting data for " +
+            "targeting hook " + sVarName + " on " + GetName(oPC));
+        return;
+    }
+
+    if (GetIsObjectValid(th.oPC))
+        _EnterTargetingMode(th.oPC, th.nObjectType, th.nHookID, nBehavior);
+}
```

Comment:
Yup, that's a good idea. I'll implement on the next commit.

Reply:
This wasn't quite as easy as I thought because of all the internal calls to re-enter targeting mode after each target is selected, so I added it as a part of the hook data added via AddTargetingHook().